File: Sorting/bubble_sort/bubble_sort_v3.py



def bubbleSort(arr):
    for i in range(len(arr) - 1):
        swap = False
        for j in range(len(arr) - 1):
            print(arr, arr[j], arr[j + 1])
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swap = True
        
        if not swap:
            break


    return arr


print(bubbleSort([12, 1, 7, 5]))

File: Sorting/bubble_sort/bubble_sort_v3.js



function bubbleSort(arr){
    let swap;
    for(let i = arr.length; i > 0; i--){
        swap = false;
        for(let j = 0; j < i - 1; j++){
            console.log(arr, arr[j], arr[j + 1]);
            if(arr[j] > arr[j + 1]){
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
                swap = true;
            } 
        }

        if(swap = false) break;
    }
    return arr;
}


console.log(bubbleSort([12, 1, 7, 5]))
